---
title: RNASeq Analysis of Arabadopsis thaliana
description: "Step by step guid on RNAseq analysis"
author: Hayley Zorkic
date: '2020-11-24'
slug: []
categories: []
tags: []
draft: no
image: 'img/portfolio/heatmapRNAseq.jpg'
showonlyimage: no
---

## Analysis of RNASeq Data from Wild Type and Mutated Arabidopsis thaliana

##  Background
In this report, we are primarily interested in understanding how the differences in gene expression between the light and dark conditions vary by gene in the primary gene ontology id GO:0006090.
To this end, we will use DESeq to look for genes with significant interaction terms between genotype and timepoint.

In interpreting the RNA-seq data set, you should note the following descriptions of the different sample genotypes:

-Col wild-type Arabidopsis thaliana.
- 14Ba mutant lacking two genes for a plant specific translation initiation factor, eIFiso4G1(AT5g57870) and eIFiso4G2 (At2g24050).
- 4Ga mutant lacking the one gene for eIF4G (AT3g60240), a translation factor that is more similar to those of other eukaryotes.
   
Also note that the samples were collected from two different time points/light conditions:“End day” (light) and “Ex dark” (dark).

```{r setup, include=FALSE}
library(DESeq2)
library(tidyverse)
library(matrixStats)
library(pheatmap)
source("bio321g_rnaseq_utils.R")
```

### 1.  DESeq
A DESeqDataSet object was constructed on the gene expression data using the DESeq package to perform a hypothesis test specifically for the significance of the interaction term time:genotype in the model. 

```{r, include=FALSE}
      ## first initialize DESeq object:
dds <- DESeqDataSetFromMatrix(
      ## countData argument should be numeric matrix (or data.frame with
      ## only numeric columns which can be coerced to numeric matrix):
countData = rnaCounts,
      ## colData argument should be data.frame with grouping information
      ## (and any other non-gene-expression covariates that might be
      ## useful in modeling the counts) whose rows correspond to the
      ## columns (hence name colData) of countData:
colData = sampleAnnotation,
      ## note tilde preceding the name group of the relevant column in
      ## the data.frame sampleAnnotation provided as the colData argument
      ## in the design argument---design argument must be formula object,
      ## similar to the right-hand side of the formulas used for linear
      ## modeling with lm:
design = ~ time + genotype + time:genotype)

      ## now run DESeq analysis pipeline on object dds in order to perform a hypothesis test specifically for the significance of the interaction term 
      ## time:genotype in the model.
dds <-  DESeq(dds, test = "LRT", reduced = ~ time + genotype)
      ##Finally, use the results function from the DESeq2 package to extract a table of results for each gene from the DESeqDataSet object.
      ## sort dds results data.frame by p-value:
ddsresults <- results(dds)
ddsresults <- ddsresults[order(ddsresults$pvalue), ]

ddsresults

FDR <- sum(ddsresults$padj < 0.1, na.rm=TRUE ) 
FDR

FP <- FDR*.1
FP
```

1a. 1429 genes from the full rnacounts data are considered to have significant time:geneotype interaction terms to keep the False Discovery Rate at 10%. 

1b. 142.9 of our our list of genes with evidence of significant interactions would be considered False Positives. 

### 2. Extract the Normalized Counts from the DESeqDataSet Object 
The data was normalized with DESeq and log-transformed with an offset of 1.

```{r, include=FALSE}
## use counts function with normalized arg set to TRUE to extract
## "normalized counts" (which are not integers) from dds
## (note: counts *function* from DESeq2 package is different object
## in R than counts *data.frame* we loaded from tsv file;
## R can tell based on context which one we mean):
normed <- counts(dds, normalized=TRUE)
## save the normalized counts matrix to a tsv file:
lgNorm <- log2(normed + 1)
```

### 3. Generate A Principal Components Analysis (PCA) Plot of Normalized Gene Expression Data. 
```{R, include=FALSE}
sampleAnnotation = data.frame(
      ## set row.names of sampleAnnotation to match col.names of normed:
row.names = colnames(normed),
      ## use gsub to remove single digit \\d right before end $
      ## of sample names to get group name for each sample:
group = gsub(".$", "", colnames(normed))
)
pca = prcomp(t(lgNorm))
      ## perform PCA on Normalized data
pcaFit = rowMeans(lgNorm) + t( pca$x %*% t(pca$rotation) )
      ## have to transpose pca$x %*% t(pca$rotation) above b/c lgNorm is t(z)
      ## set up data.frame pcaData for ggplot...
pcaData = data.frame(pca$x[ , 1:2])
      ## add in sample annotation info
pcaData$group = sampleAnnotation[rownames(pcaData), "group"]
      ## and sample names
pcaData$sample = rownames(pcaData)

gg = ggplot(pcaData, aes(x=PC1, y=PC2, color = group, label=sample)) + 
  geom_point(size=2.5, alpha=0.75) + 
  scale_color_manual(values=groupColors) + 
  ggtitle("Principal Components Analysis (PCA) of Arabidopsis thaliana RnaSeq Data") +
  theme_light()
```

```{R, fig.height=5, fig.width=10, echo=FALSE}
gg
```
A gene expression PCA plot was created to provide insight into the association between samples. In the PC2 principle cluster direction, the 14Ba mutants are most separated from both the 4G mutants and Wild Type (Col) which are closly clustered. In the PC1 principle cluster direction, the dark and light samples were separated together in all 3 sample groups. 

### 4. Extracting Gene Expression Data from Genes with the Primary Gene Ontology Id GO:0006090

Using R, a vector containing the ids of the genes associated with the primary gene ontology id GO:0006090 was extracted containing three columns: 
  (1) gene id, 
  (2) gene name/symbol, and 
  (3) gene description. 
This file has been provided and is named "GO0006090geneset.tsv".

```{r, include=FALSE}
      ##read in gene_sets.tsv file
genesets = read.table("gene_sets.tsv.gz",
                       sep="\t",header=TRUE,
                       quote="", comment.char="")
      ## grab only those rows with a gene ontology id of GO:0006090
GO0006090 <- genesets %>% filter(gene_ontology_primary_id == "GO:0006090")

      ## make a new data.frame containing the geneID, name/symbol, and descriptions of those genes that are included in the gene ontology id GO:0006090
      ## geneNamesAndDescriptions: data.frame with rownames corresponding to gene
      ## ids and three columns:
      ## (1) gene :: gene id (same as rownames,
      ## (2) symbol :: gene name/symbol
      ## (3) description :: gene description
geneset <- inner_join(GO0006090, geneNamesAndDescriptions)
      ## select relevant columns
geneset <- data.frame(geneset[,c(4:6)])
      ## rename columns
names(geneset) <- c("geneID", "name/symbol","descirption")
      ## write the new data.frame to a .tsv file

geneset
write.table(data.frame(geneset),"GO0006090geneset.tsv", sep = "\t", row.names = FALSE, quote = FALSE)
```

### 5. Normalizing GO:0006090 Genes

By filtering lgNorm to retain only those rows of that data.frame corresponding to genes in GO:0006090, a new data.frame was made.

```{r, include=FALSE}
lgNorm <- log2(normed + 1)
lgNorm <- data.frame(lgNorm)
      ## make sure lgNorm is a data.frame
      ## make the rownames a column name
lgNorm <- rownames_to_column(lgNorm, var = "gene")
      ## make a new data.frame containing only the rows of normalized data that correspond to genes with a gene ontology id of GO:0006090
lgGo <- lgNorm %>% filter(lgNorm[,1] %in% geneset$geneID)
      ## get rid of the X's in front of the column names... I know there is a better way to do this, revisit 
names(lgGo) <- c("gene","14BENDDAY2", "14BENDDAY4", "14BEXDARK2", "14BEXDARK3", "14BEXDARK4", "4GENDDAY2", "4GENDDAY3", "4GENDDAY4", "4GEXDARK2", "4GEXDARK3" , "COLENDDAY3" , "COLENDDAY5" , "COLEXDARK2" , "COLEXDARK3" , "COLEXDARK4")
      ## remove gene column- make it row names
lgGo <- column_to_rownames(lgGo, var = "gene")
```

### 6. Generate A Principal Components Analysis (PCA) Plot of Normalized Gene Expression Data from GO:0006090.
  
```{r, include=FALSE}
## use prcomp function for PCA in R
      ## on transposed version of lgNorm, t(lgNorm),
      ## since prcomp assumes variables (genes) are in columns, not rows
      ## z_ig = lgNorm[g, i] = t(lgNorm)[i, g]
pca = prcomp(t(lgGo))

## set up data.frame pcaData for ggplot...
pcaData = data.frame(pca$x[ , 1:2])

## add in sample annotation info
pcaData$group = sampleAnnotation[rownames(pcaData), "group"]

## and sample names
pcaData$sample = rownames(pcaData)

pcaData

## make sure ggplot2 library is loaded
gg = ggplot(pcaData, aes(x=PC1, y=PC2, color=group)) +
  geom_point(size=2.5, alpha=0.75) +
  scale_color_manual(values=groupColors)+
  ggtitle("Principal Components Analysis (PCA) of Arabidopsis thaliana RnaSeq Data 
GO:0006090") + 
  theme_light()
```

```{R, fig.height=5, fig.width=10, echo=FALSE}
gg
```

A gene expression PCA plot was created to provide insight into the association between samples within the same gene ontology. In the PC2 principle cluster direction, the 14Ba mutants are most separated from both the 4G mutants and Wild Type (Col) which are closly clustered. In the PC1 principle cluster direction, the dark and light samples were separated together in all 3 sample groups.These results are congruent with the overall gene expression data. 

### 7. Gene Expression Heatmap of Normalized Gene Expression Data from GO:0006090.
```{R, include=FALSE}
heatData = lgGo - rowMeans(lgGo)

heatData[heatData > 2] = 2; heatData[heatData < -2] = -2

```

```{R, fig.height=15, fig.width=10, echo=FALSE}
pheatmap(
heatData,
color = heatPalette,
clustering_method = "average",
labels_row=geneNamesAndDescriptions[rownames(heatData), "symbol"]
)

```

From the sample-axis dendrogram, we can see thee are two major groups- one with the samples from light and another with samples from DARK. The gene expression patterns between these two groups are inversly related upon visual inspection of the heatmap. 

From the gene-axis dendrogram, we can see that there is a division between the gene group {NADP-ME1, AT1G17410, HDR, and AT1G18270, and AT3G49160} and the rest of the genes in the gene ontology. The gene group of interst have much higher diffeences in expression values from the means of the respective gene. 

The gene TSO2, not apart of our gene group of interest, also appears to have strong deviation from the mean gene expression levels. 


### 8. Close Look at Top 9 Most Significant time:genotype Interaction. 

A stripchart was generated to show the expression levels of these genes by time and genotype with the must significant time:genotype interaction.

```{r, echo=FALSE}
sampleAnnotation = read.table("rna_sample_annotation.tsv",
sep="\t", header=TRUE, row.names=1, check.names=FALSE)
data <- data.frame(ddsresults)
lgGo <- rownames_to_column(lgGo, var="gene")
data <- rownames_to_column(data, var="gene")
genes <- data %>% filter(data$gene %in% lgGo$gene)%>% arrange(pvalue) %>% top_n(9)
lgGo <- lgGo %>% filter(lgGo$gene %in% genes$gene)
lgGo <- column_to_rownames(lgGo, "gene")

gg <- stripchart321g(lgGo, sampleAnnotation)
gg
```
From the stripchart we can see that for the genes AT1G09780, AT3G23580, AT1G12000, AT1G74030, AT2G22480, AT4G09320, expression levels in all three genotypes (14B, 4G, COL) were higher during the light compared to dark. Inversely, for the genes AT1G12900 the expression levels in all three genotypes (14B, 4G, COL) were lower during the light compared to dark. 

In all 3 of the genotypes, gene expression in the genes AT3G25860, AT1G30120 is not significantly different between light and dark times but is significant between distinct between genotypes; 14B is higher than COL(wt) which is higher than 4G. 

4Ga is the mutant lacking the one gene for eIF4G (AT3g60240), a translation factor that is moresimilar to those of other eukaryotes. In this genotype, we can see that there is a clear difference in gene expression during the light versus the dark following the general expression patterns stated above. The 4G genotype had lower gene expression levels than wild type during light AND dark in every gene except for AT2G22480, and potentially in AT1G74030 where wild type light expression is lower than 4G light expression and wild type dark expression was lower than 4G dark expression.  

14B is the mutant lacking two genes for a plant specific translation initiation factor, eIFiso4G1(AT5g57870) and eIFiso4G2 (At2g24050).In this genotype, we can see that there is a clear difference in gene expression during the light versus the dark following the general expression patterns stated above. The 14B genotype had higher gene expression levels than wild type during light AND dark in every gene except for AT2G22480, and potentially in AT1G74030 where wild type light expression was higher than 14B light expression and wild type dark expression was higher than 14B dark expression.  

COL is the wild type. In this genotype, we can see that there is a clear difference in gene expression during the light versus the dark following the general expression patterns stated above. To summarize the gene expression observed in the stripchart above, the 14B mutant increaed expression in light and dark conditions compared to wild type and the 4G mutant decreased expression in light and dark conditions compared to wild type. 






