<!DOCTYPE html>
<html lang="en-us">
<head><meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<title>HAYLEY ZORKIC</title>
<meta name="description" content="The Personal Website of Hayley Zorkic">
<meta name="viewport" content="width=device-width, initial-scale=1">
<link rel="stylesheet" href="https://hzorkic.github.io/css/bootstrap.min.css">
<link rel="stylesheet" href="//fonts.googleapis.com/css?family=Roboto:400,300,700,400italic">
<link rel="stylesheet" href="https://hzorkic.github.io/css/font-awesome.min.css">
<link rel="stylesheet" href="https://hzorkic.github.io/css/owl.carousel.css">
<link rel="stylesheet" href="https://hzorkic.github.io/css/owl.theme.css">


  <link href="https://hzorkic.github.io/css/style.default.css" rel="stylesheet" id="theme-stylesheet">

 

  
    <!--[if lt IE 9]>
        <script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
        <script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
    <![endif]-->
  


<link href="https://hzorkic.github.io/css/custom.css" rel="stylesheet">
<link rel="shortcut icon" href="https://hzorkic.github.io/img/aobut.png">


<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
	ga('create', 'G-EGR7PE3Y44', 'auto');
	
	ga('send', 'pageview');
}
</script>
<script async src='https://www.google-analytics.com/analytics.js'></script>

</head>
<body>
  <div id="all">
      <div class="container-fluid">
          <div class="row row-offcanvas row-offcanvas-left">
              <div id="sidebar" class="col-xs-6 col-sm-4 col-md-3 sidebar-offcanvas">
  <div class="sidebar-content">
    
    <img style="border: solid 2px black; border-radius: 50%; max-width: 90%; height: auto;" src="/img/1_about.jpg" alt="">
    <h1></h1>
    
    <h1 class="sidebar-heading"><a href="https://hzorkic.github.io/">HAYLEY ZORKIC</a></h1>
    
      <p class="sidebar-p">Data Scientist.</p>
    
    
    <ul class="sidebar-menu">
      
        <li><a href="https://hzorkic.github.io/about/">About</a></li>
      
        <li><a href="https://hzorkic.github.io/portfolio/">Posts</a></li>
      
        <li><a href="https://hzorkic.github.io/img/RESUME.pdf">CV</a></li>
      
    </ul>
    
    <p class="social">
  
  
  
  
  
  <a href="mailto:hayley.zorkic@gmail.com" data-animate-hover="pulse" class="email">
    <i class="fa fa-envelope"></i>
  </a>
  
  
  <a href="www.linkedin.com/in/hayley-zorkic" data-animate-hover="pulse" class="external">
    <i class="fa fa-linkedin"></i>
  </a>
  
  
  
  <a href="https://github.com/hzorkic" data-animate-hover="pulse" class="external">
    <i class="fa fa-github"></i>
  </a>
  
  
  
  
</p>

  </div>
</div>

              
<div class="col-xs-12 col-sm-8 col-md-9 content-column white-background">
  <div class="small-navbar visible-xs ">
  
  <button type="button" data-toggle="offcanvas" class="btn btn-ghost pull-left">
    <i class="fa fa-bars"></i>
  </button>

  <h1 class="small-navbar-heading"><a href="https://hzorkic.github.io/">HAYLEY ZORKIC</a></h1>
  
</div>


  <div class="row">
    <div class="col-lg-8">
      <div class="content-column-content">
         <h1>RNASeq Analysis of Arabadopsis thaliana</h1>
         
<script src="https://hzorkic.github.io/portfolio/2020-11-24-rnaseq-analysis-of-arabadopsis-thaliana/index_files/header-attrs/header-attrs.js"></script>


<div id="analysis-of-rnaseq-data-from-wild-type-and-mutated-arabidopsis-thaliana" class="section level2">
<h2>Analysis of RNASeq Data from Wild Type and Mutated Arabidopsis thaliana</h2>
<div id="background" class="section level3">
<h3>0. Background</h3>
<p>In this report, we are primarily interested in understanding how the differences in gene expression between the light and dark conditions vary by gene in the primary gene ontology id <a href="GO:0006090" class="uri">GO:0006090</a>.
To this end, we will use DESeq to look for genes with significant interaction terms between genotype and timepoint.</p>
<p>In interpreting the RNA-seq data set, you should note the following descriptions of the different sample genotypes:</p>
<p>-Col wild-type Arabidopsis thaliana.
- 14Ba mutant lacking two genes for a plant specific translation initiation factor, eIFiso4G1(AT5g57870) and eIFiso4G2 (At2g24050).
- 4Ga mutant lacking the one gene for eIF4G (AT3g60240), a translation factor that is more similar to those of other eukaryotes.</p>
<p>Also note that the samples were collected from two different time points/light conditions:“End day” (light) and “Ex dark” (dark).</p>
</div>
<div id="deseq" class="section level3">
<h3>1. DESeq</h3>
<p>A DESeqDataSet object was constructed on the gene expression data using the DESeq package to perform a hypothesis test specifically for the significance of the interaction term time:genotype in the model.</p>
<p>1a. 1429 genes from the full rnacounts data are considered to have significant time:geneotype interaction terms to keep the False Discovery Rate at 10%.</p>
<p>1b. 142.9 of our our list of genes with evidence of significant interactions would be considered False Positives.</p>
</div>
<div id="extract-the-normalized-counts-from-the-deseqdataset-object" class="section level3">
<h3>2. Extract the Normalized Counts from the DESeqDataSet Object</h3>
<p>The data was normalized with DESeq and log-transformed with an offset of 1.</p>
</div>
<div id="generate-a-principal-components-analysis-pca-plot-of-normalized-gene-expression-data." class="section level3">
<h3>3. Generate A Principal Components Analysis (PCA) Plot of Normalized Gene Expression Data.</h3>
<p><img src="https://hzorkic.github.io/portfolio/2020-11-24-rnaseq-analysis-of-arabadopsis-thaliana/index_files/figure-html/unnamed-chunk-4-1.png" width="960" />
A gene expression PCA plot was created to provide insight into the association between samples. In the PC2 principle cluster direction, the 14Ba mutants are most separated from both the 4G mutants and Wild Type (Col) which are closly clustered. In the PC1 principle cluster direction, the dark and light samples were separated together in all 3 sample groups.</p>
</div>
<div id="extracting-gene-expression-data-from-genes-with-the-primary-gene-ontology-id-go0006090" class="section level3">
<h3>4. Extracting Gene Expression Data from Genes with the Primary Gene Ontology Id <a href="GO:0006090" class="uri">GO:0006090</a></h3>
<p>Using R, a vector containing the ids of the genes associated with the primary gene ontology id <a href="GO:0006090" class="uri">GO:0006090</a> was extracted containing three columns:
(1) gene id,
(2) gene name/symbol, and
(3) gene description.
This file has been provided and is named “GO0006090geneset.tsv”.</p>
</div>
<div id="normalizing-go0006090-genes" class="section level3">
<h3>5. Normalizing <a href="GO:0006090" class="uri">GO:0006090</a> Genes</h3>
<p>By filtering lgNorm to retain only those rows of that data.frame corresponding to genes in <a href="GO:0006090" class="uri">GO:0006090</a>, a new data.frame was made.</p>
</div>
<div id="generate-a-principal-components-analysis-pca-plot-of-normalized-gene-expression-data-from-go0006090." class="section level3">
<h3>6. Generate A Principal Components Analysis (PCA) Plot of Normalized Gene Expression Data from <a href="GO:0006090" class="uri">GO:0006090</a>.</h3>
<p><img src="https://hzorkic.github.io/portfolio/2020-11-24-rnaseq-analysis-of-arabadopsis-thaliana/index_files/figure-html/unnamed-chunk-8-1.png" width="960" /></p>
<p>A gene expression PCA plot was created to provide insight into the association between samples within the same gene ontology. In the PC2 principle cluster direction, the 14Ba mutants are most separated from both the 4G mutants and Wild Type (Col) which are closly clustered. In the PC1 principle cluster direction, the dark and light samples were separated together in all 3 sample groups.These results are congruent with the overall gene expression data.</p>
</div>
<div id="gene-expression-heatmap-of-normalized-gene-expression-data-from-go0006090." class="section level3">
<h3>7. Gene Expression Heatmap of Normalized Gene Expression Data from <a href="GO:0006090" class="uri">GO:0006090</a>.</h3>
<p><img src="https://hzorkic.github.io/portfolio/2020-11-24-rnaseq-analysis-of-arabadopsis-thaliana/index_files/figure-html/unnamed-chunk-10-1.png" width="960" /></p>
<p>From the sample-axis dendrogram, we can see thee are two major groups- one with the samples from light and another with samples from DARK. The gene expression patterns between these two groups are inversly related upon visual inspection of the heatmap.</p>
<p>From the gene-axis dendrogram, we can see that there is a division between the gene group {NADP-ME1, AT1G17410, HDR, and AT1G18270, and AT3G49160} and the rest of the genes in the gene ontology. The gene group of interst have much higher diffeences in expression values from the means of the respective gene.</p>
<p>The gene TSO2, not apart of our gene group of interest, also appears to have strong deviation from the mean gene expression levels.</p>
</div>
<div id="close-look-at-top-9-most-significant-timegenotype-interaction." class="section level3">
<h3>8. Close Look at Top 9 Most Significant time:genotype Interaction.</h3>
<p>A stripchart was generated to show the expression levels of these genes by time and genotype with the must significant time:genotype interaction.</p>
<pre><code>## Selecting by padj</code></pre>
<p><img src="https://hzorkic.github.io/portfolio/2020-11-24-rnaseq-analysis-of-arabadopsis-thaliana/index_files/figure-html/unnamed-chunk-11-1.png" width="672" />
From the stripchart we can see that for the genes AT1G09780, AT3G23580, AT1G12000, AT1G74030, AT2G22480, AT4G09320, expression levels in all three genotypes (14B, 4G, COL) were higher during the light compared to dark. Inversely, for the genes AT1G12900 the expression levels in all three genotypes (14B, 4G, COL) were lower during the light compared to dark.</p>
<p>In all 3 of the genotypes, gene expression in the genes AT3G25860, AT1G30120 is not significantly different between light and dark times but is significant between distinct between genotypes; 14B is higher than COL(wt) which is higher than 4G.</p>
<p>4Ga is the mutant lacking the one gene for eIF4G (AT3g60240), a translation factor that is moresimilar to those of other eukaryotes. In this genotype, we can see that there is a clear difference in gene expression during the light versus the dark following the general expression patterns stated above. The 4G genotype had lower gene expression levels than wild type during light AND dark in every gene except for AT2G22480, and potentially in AT1G74030 where wild type light expression is lower than 4G light expression and wild type dark expression was lower than 4G dark expression.</p>
<p>14B is the mutant lacking two genes for a plant specific translation initiation factor, eIFiso4G1(AT5g57870) and eIFiso4G2 (At2g24050).In this genotype, we can see that there is a clear difference in gene expression during the light versus the dark following the general expression patterns stated above. The 14B genotype had higher gene expression levels than wild type during light AND dark in every gene except for AT2G22480, and potentially in AT1G74030 where wild type light expression was higher than 14B light expression and wild type dark expression was higher than 14B dark expression.</p>
<p>COL is the wild type. In this genotype, we can see that there is a clear difference in gene expression during the light versus the dark following the general expression patterns stated above. To summarize the gene expression observed in the stripchart above, the 14B mutant increaed expression in light and dark conditions compared to wild type and the 4G mutant decreased expression in light and dark conditions compared to wild type.</p>
</div>
</div>

         
      </div>
    </div>
  </div>
</div>

          </div>
      </div>
  </div>
  <script src="https://hzorkic.github.io/js/jquery.min.js"></script>
<script src="https://hzorkic.github.io/js/bootstrap.min.js"></script>
<script src="https://hzorkic.github.io/js/jquery.cookie.js"> </script>
<script src="https://hzorkic.github.io/js/ekko-lightbox.js"></script>
<script src="https://hzorkic.github.io/js/jquery.scrollTo.min.js"></script>
<script src="https://hzorkic.github.io/js/masonry.pkgd.min.js"></script>
<script src="https://hzorkic.github.io/js/imagesloaded.pkgd.min.js"></script>
<script src="https://hzorkic.github.io/js/owl.carousel.min.js"></script>
<script src="https://hzorkic.github.io/js/front.js"></script>



</body>
</html>
